File: backend/pkg/models/variable.go
package models

import "time"

type Variable struct {
	Id          int64     `json:"id"`
	Name        string    `json:"name"`
	Type        string    `json:"type"`
	Value       string    `json:"value"`
	Datasource  int       `json:"datasource"`
	Desc        string    `json:"description"`
	Created     time.Time `json:"created"`
	Refresh     string    `json:"refresh"`
	EnableMulti bool      `json:"enableMulti"`
	EnableAll   bool      `json:"enableAll"`
}

File: datav/frontend/src/types/variable.ts
// Copyright 2023 xObserve.io Team

export interface Variable {
  id?: number
  name: string
  type: string
  datasource?: number
  value?: string // query statement
  default?: string
  description?: string
  created?: string
  values?: string[]
  selected?: string
  regex?: string
  refresh?: VariableRefresh
  enableMulti?: boolean
  enableAll?: boolean
  sortWeight?: number
  teamId?: number
}

export enum VariableQueryType {
  Custom = 'custom',
  Query = 'query',
  Datasource = 'datasource',
  TextInput = 'textinput',
}

export enum VariableRefresh {
  OnDashboardLoad = 'OnDashboardLoad',
  OnTimeRangeChange = 'OnTimeRangeChange',
  Manually = 'Manually',
}
